### PR TITLE
Add validation plots checking packet fraction sums

### PR DESCRIPTION
**Pull request by diaza**

### CODE REVIEW

File: run-validation/flow_validation.py [hunk 1]
```diff
@@ -578,6 +578,19 @@ def main(flow_file, charge_only):
             output.savefig()
             plt.close()
 
+        # Check the sums of the fraction field for charge deposition. They should add to 1.
+        if 'mc_truth' in flow_h5.keys():
+            fractions = flow_h5['mc_truth/packet_fraction/data']['fraction']
+            summed_fractions = fractions.sum(axis=-1)
+            fig, ax = plt.subplots(constrained_layout = True)
+            ax.hist(summed_fractions, bins = np.arange(0.895,1.1,0.01), density = True)
```
Comment (krwood) [inline run-validation/flow_validation.py]: What if something is terribly wrong and the fraction is outside of the hardcoded range?...

File: run-validation/larndsim_validation.py [hunk 2]
```diff
@@ -174,6 +174,17 @@ def main(sim_file, charge_only):
         plt.ylabel("Counts")
         output.savefig()
         plt.close()     
+
+        # Check the sums of the fraction field for charge deposition. They should add to 1.
+        fractions = mc_packets_assn['fraction']
+        summed_fractions = fractions.sum(axis=-1)
+        fig, ax = plt.subplots(constrained_layout = True)
+        ax.hist(summed_fractions, bins = np.arange(0.895,1.1,0.01), density = True)
```
Comment (krwood) [inline run-validation/larndsim_validation.py]: ^^

File: run-validation/flow_validation.py (hunk 1 above)
Comment (diaza) [reply run-validation/flow_validation.py]: Yee, I've expanded the range to include everything and changed the y-axis to log scale.

Comment (krwood) [reply run-validation/flow_validation.py]: What about just specifying the number of bins with `bins = 250` (or whatever) and letting the plotter autorange?

File: run-validation/flow_validation.py [hunk 3]
```diff
@@ -583,10 +583,11 @@ def main(flow_file, charge_only):
             fractions = flow_h5['mc_truth/packet_fraction/data']['fraction']
             summed_fractions = fractions.sum(axis=-1)
             fig, ax = plt.subplots(constrained_layout = True)
-            ax.hist(summed_fractions, bins = np.arange(0.895,1.1,0.01), density = True)
+            ax.hist(summed_fractions, bins= np.arange(-0.05, summed_fractions.max(), 0.1))
```
Comment (diaza) [inline run-validation/flow_validation.py:586]: @krwood Re: previous comment
Not sure if you saw the update, but here I changed here to have the upper bound be determined by the maximum value, so it should capture the largest value.

**Review by krwood — APPROVED** [review]

**Pull request merged**